test/python/quantum_info/operators/symplectic/test_pauli_list.py

```diff
@@ -2058,20 +2058,55 @@ def qubitwise_commutes(left: Pauli, right: Pauli) -> bool:
 
         # checking that every input Pauli in pauli_list is in a group in the ouput
         output_labels = [pauli.to_label() for group in groups for pauli in group]
-        assert sorted(output_labels) == sorted(input_labels)
+        #     assert sorted(output_labels) == sorted(input_labels)
+        self.assertListEqual(sorted(output_labels), sorted(input_labels))
+        # Within each group, every operator qubit-wise commutes with every other operator.
+        for group in groups:
+            self.assertTrue(
+                all(
+                    qubitwise_commutes(pauli1, pauli2)
+                    for pauli1, pauli2 in itertools.combinations(group, 2)
+                )
+            )
+        # For every pair of groups, at least one element from one does not qubit-wise commute with
+        # at least one element of the other.
+        for group1, group2 in itertools.combinations(groups, 2):
+            self.assertFalse(
+                all(
+                    qubitwise_commutes(group1_pauli, group2_pauli)
+                    for group1_pauli, group2_pauli in itertools.product(group1, group2)
+                )
+            )
 
+    def test_group_commuting(self):
+        """Test general grouping commuting operators"""
+
+        def commutes(left: Pauli, right: Pauli) -> bool:
+            return len(left) == len(right) and left.commutes(right)
+
+        input_labels = ["IY", "ZX", "XZ", "YI", "YX", "YY", "YZ", "ZI", "ZX", "ZY", "iZZ", "II"]
+        np.random.shuffle(input_labels)
+        pauli_list = PauliList(input_labels)
+        #  if qubit_wise=True, equivalent to test_group_qubit_wise_commuting
+        groups = pauli_list.group_commuting(qubit_wise=False)
+
+        # checking that every input Pauli in pauli_list is in a group in the ouput
+        output_labels = [pauli.to_label() for group in groups for pauli in group]
+        #     assert sorted(output_labels) == sorted(input_labels)
+        self.assertListEqual(sorted(output_labels), sorted(input_labels))
         # Within each group, every operator qubit-wise commutes with every other operator.
         for group in groups:
-            assert all(
-                qubitwise_commutes(pauli1, pauli2)
-                for pauli1, pauli2 in itertools.combinations(group, 2)
+            self.assertTrue(
+                all(commutes(pauli1, pauli2) for pauli1, pauli2 in itertools.combinations(group, 2))
             )
```

Comment:
Please be careful with the comments - it's vitally important that they are accurate to the code to help future maintainers understand what's going on, especially in tests and mathematical code.  Copy-pasting code is generally not a good idea, because it leads to mistakes like this.

There are a few other places in this PR at the moment where there are copy-pasted comments that aren't accurate to the surrounding code.  Please could you check through all the comments carefully, and make sure that they're accurate?  It might help to use the PR review panel on GitHub to show all the changes you've made, so you can check that the PR is in the form you intend it to be in.